internal/driver/readhandler.go

```diff
@@ -157,11 +235,91 @@ func (d *Driver) makeMethodCall(deviceClient *opcua.Client, req sdkModel.Command
 	if resp.StatusCode != ua.StatusOK {
 		return nil, fmt.Errorf("Driver.handleReadCommands: Method status not OK: %v", resp.StatusCode)
 	}
-	result, err := newResult(req, resp.OutputArguments[0].Value())
 
-	// get source timestamp
-	sourceTimeStamp := extractSourceTimestamp(resp.OutputArguments[0].DataValue())
-	result.Tags["source timestamp"] = sourceTimeStamp.String()
+	return newResult(req, resp.OutputArguments[0].Value())
+}
 
-	return result, err
+func generateCert() (*tls.Certificate, error) {
```

Comment:
Impressive work! I can not review if it's correct and appropriate in detail. Have to trust you.